### PR TITLE
Split Datamodel Parsing and Formatting

### PR DESCRIPTION
**Pull request by do4gr**
This separates the logic for parsing and formatting. 
Parsing should only add relation names to the datamodel and enrich many to many relations. 
Formatting will enrich all relations (fields and references arguments), will add back-relations and add underlying scalar fields.

### CODE REVIEW

File: libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline.rs [hunk 1]
```diff
@@ -64,11 +67,17 @@ impl<'a, 'b> ValidationPipeline<'a> {
...
 
         // TODO: Move consistency stuff into different module.
-        // Phase 5: Consistency fixes. These don't fail.
-        if let Err(mut err) = self.standardiser.standardise(ast_schema, &mut schema) {
+        // Phase 5: Consistency fixes. These don't fail and always run, during parsing AND formatting
+        if let Err(mut err) = self.standardiser_for_parsing.standardise(&mut schema) {
             diagnostics.append(&mut err);
         }
 
+        // Phase 5 formatting: Consistency fixes. These only run during formatting.
```
Comment (tomhoule) [inline libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline.rs:75]: note from zoom call: we have two times phase 5

File: libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline.rs [hunk 2]
```diff
@@ -64,11 +67,17 @@ impl<'a, 'b> ValidationPipeline<'a> {
...
+        if transform {
+            if let Err(mut err) = self.standardiser_for_formatting.standardise(ast_schema, &mut schema) {
+                diagnostics.append(&mut err);
```
Comment (tomhoule) [inline libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline.rs:78]: We don't really know what happens when a reformatter-specific validation error happens

File: libs/datamodel/core/src/transform/ast_to_dml/standardise_parsing.rs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+impl StandardiserForParsing {
+    /// Creates a new instance, with all builtin attributes registered.
+    pub fn new() -> Self {
+        StandardiserForParsing {}
+    }
+
+    pub fn standardise(&self, schema: &mut dml::Datamodel) -> Result<(), Diagnostics> {
+        self.name_unnamed_relations(schema);
+        self.set_relation_to_field_to_id_if_missing(schema);
```
Comment (tomhoule) [inline libs/datamodel/core/src/transform/ast_to_dml/standardise_parsing.rs:17]: doc/comment to be added that this is only for m2m relations

File: libs/datamodel/core/tests/attributes/relations/relations_negative.rs [hunk 4]
```diff
@@ -0,0 +1,551 @@
...
+
+    let errors = parse_error(dml);
+
+    errors.assert_is(
+        DatamodelError::new_field_validation_error("The relation field `createdBy` on Model `Code` is missing an opposite relation field on the model `User`. Either run `prisma format` or add it manually.", "Code", "createdBy",Span::new(103, 121)),
+    );
+}
+
+#[test]
+fn must_add_referenced_fields_on_both_sides_for_one_to_many_relations() {
```
Comment (tomhoule) [inline libs/datamodel/core/tests/attributes/relations/relations_negative.rs:492]: the test name should be adjusted, it errors now (as it should)

File: libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs [hunk 5]
```diff
@@ -325,6 +326,615 @@ model Post {
     assert_reformat(input, expected);
 }
 
+#[test]
+#[ignore]
```
Comment (tomhoule) [inline libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs:330]: noticed during zoom call: should this be ignored?

File: libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs [hunk 6]
```diff
@@ -325,6 +326,615 @@ model Post {
...
+      userId Int
+      user   User @relation(references: [id], fields: [userId])
+    }
+    "#};
+
+    assert_reformat(input, expected);
+}
+
+#[test]
+fn must_add_referenced_fields_on_both_sides_for_one_to_many_relations() {
```
Comment (tomhoule) [inline libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs:570]: from zoom call: check the name

File: libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs [hunk 7]
```diff
@@ -362,42 +972,34 @@ model Post {
...
-"#;
-    assert_reformat(input, expected);
+    assert_eq!(input, result);
 }
 
 fn assert_reformat(schema: &str, expected_result: &str) {
-    println!("schema: {:?}", schema);
     let result = datamodel::ast::reformat::Reformatter::new(&schema).reformat_to_string();
-    println!("result: {}", result);
     assert_eq!(result, expected_result);
```
Comment (tomhoule) [inline libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs:1004]: zoom comment: maybe we could validate the final result in this method? (or another?)

File: libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline.rs (hunk 2 above)
Comment (do4gr) [reply libs/datamodel/core/src/transform/ast_to_dml/validation_pipeline.rs:78]: They mostly get ignored, we'll revisit that later.

**Pull request merged**